spec/project_spec.rb

```diff
@@ -723,18 +723,36 @@ class InvalidErrorHandler; end
             allow(project_instance.decision_service).to receive(:get_variation_for_feature).and_return(decision_to_return)
 
             expect(project_instance.get_feature_variable_string('boolean_single_variable_feature', 'boolean_variable', user_id, user_attributes))
-              .to eq('true')
+              .to eq(nil)
 
-            expect(spy_logger).to have_received(:log).twice
+            expect(spy_logger).to have_received(:log).once
             expect(spy_logger).to have_received(:log).once
               .with(
-                Logger::INFO,
-                "Got variable value 'true' for variable 'boolean_variable' of feature flag 'boolean_single_variable_feature'."
+                Logger::WARN,
+                "Requested variable type 'string' but variable 'boolean_variable' is of type 'boolean'."
               )
+          end
+        end
+
+        describe 'and the variable type integer is not a string' do
+          it 'should log a warning' do
+            integer_feature = project_instance.config.feature_flag_key_map['integer_single_variable_feature']
+            experiment_to_return = project_instance.config.experiment_id_map[integer_feature['experimentIds'][0]]
+            variation_to_return = experiment_to_return['variations'][0]
+            decision_to_return = {
+                'experiment' => experiment_to_return,
+                'variation' => variation_to_return
+            }
+            allow(project_instance.decision_service).to receive(:get_variation_for_feature).and_return(decision_to_return)
+
+            expect(project_instance.get_feature_variable_string('integer_single_variable_feature', 'integer_variable', user_id, user_attributes))
+              .to eq(nil)
+
+            expect(spy_logger).to have_received(:log).once
```

Comment:
Why two lines with?
expect(spy_logger).to have_received(:log).once

Reply:
Done!